# kf2vec_data 
Data and scripts used for validation of kf2vec software at https://github.com/noraracht/kf2vec.

Populating...

## Raw data
* Models (uplodad in progress)
* Input k-mer frequencies (uplodad in progress)


## Summary data

<!---This section contains summary data tables and scripts we used to process them.--->

* Placement of full genomes
  
  * Model parameters, effects of divide-and-conquer, and chunking
    - The directory [full_genomes](https://github.com/noraracht/kf2vec_data/tree/main/full_genomes) contains results for the placement of full genomes. These summary tables serve as an input into the script [incomplete_genomes_git_full_genomes.R](https://github.com/noraracht/kf2vec_data/blob/main/full_genomes/incomplete_genomes_git_full_genomes.R) and were used to generate plots in Figures 2A, S1, S3, and S7.
      
  * Variable k-mer length
    - The directory [kmer_len](https://github.com/noraracht/kf2vec_data/tree/main/kmer_len) contains results for the variable k-mer length experiment, and the script [incomplete_genomes_git_kmerlen.R](https://github.com/noraracht/kf2vec_data/blob/main/kmer_len/incomplete_genomes_git_kmerlen.R) that was used to create Figure 2B.
  
* Effect of divide and conquer


* Benchmarking

  * Comparison to CAFE 
    - The directory [cafe_dist_comparison](https://github.com/noraracht/kf2vec_data/tree/main/cafe_dist_comparison) contains the summary results along with the script [placement_err_cafe_vs_us_git.R](https://github.com/noraracht/kf2vec_data/blob/main/cafe_dist_comparison/placement_err_cafe_vs_us_git.R) used to generate plots shown in Figures 2C and 2D.

  * Comparison to DEPP
     - The Figures 2E and 2F.


* Placement of incomplete genomes
  
  * Controlled length fragments
    - The [controlled_fragments](https://github.com/noraracht/kf2vec_data/tree/main/controlled_fragments) directory contains multiple summary result files that serve as inputs to the script [incomplete_genomes_git_fragments.R](https://github.com/noraracht/kf2vec_data/blob/main/controlled_fragments/incomplete_genomes_git_fragments.R), which was used to produce the diagrams in Figures 3A and S6.
  
  * Contigs
    - Summary results for contig placement are provided in the file [summary_TOL_full_genome_contigs_six_models.pl_error.gz](https://github.com/noraracht/kf2vec_data/blob/main/tol_contigs/summary_TOL_full_genome_contigs_six_models.pl_error.gz), which should be unarchived using `gunzip`. The resulting table serves as input to the script [placement_err_dev_queries_git_contigs.R](https://github.com/noraracht/kf2vec_data/blob/main/tol_contigs/placement_err_dev_queries_git_contigs.R), which generates the diagrams shown in Figures 3B and S8â€“9.
    
  * HiFi reads
    - Summary results for long-read placement are provided in the files [summary_hifi_six_models.pl_error_part1.csv.gz](https://github.com/noraracht/kf2vec_data/blob/main/hifi/summary_hifi_six_models.pl_error_part1.csv.gz) and [summary_hifi_six_models.pl_error_part2.csv.gz](https://github.com/noraracht/kf2vec_data/blob/main/hifi/summary_hifi_six_models.pl_error_part2.csv.gz), which should be unarchived using `gunzip`. hese files serve as input to the script [Summary_reads_placement_git_hifi.R](https://github.com/noraracht/kf2vec_data/blob/main/hifi/Summary_reads_placement_git_hifi.R), which produced Figures 3C and 3D.
    

    
* CAMI2
  - The directory [cami2](https://github.com/noraracht/kf2vec_data/tree/main/cami2) contains the CAMI2 analysis results generated by [AMBER](https://github.com/CAMI-challenge/AMBER). The script [placement_err_dev_queries_git_cami2.R](https://github.com/noraracht/kf2vec_data/blob/main/cami2/placement_err_dev_queries_git_cami2.R) was used to generate the plots shown in Figures 4A and 4B.
  
